src/perl/lib/wtsi_clarity/epp/generic/roles/barcode_common.pm

```diff
@@ -3,10 +3,18 @@ package wtsi_clarity::epp::generic::roles::barcode_common;
 use Moose::Role;
 use Readonly;
 use Carp;
+use List::Util qw(sum);
+
+local $ENV{'WTSI_CLARITY_HOME'} = q[t/data/config];
+
+use wtsi_clarity::util::config;
+my $config = wtsi_clarity::util::config->new();
+my $base_uri = $config->clarity_api->{'base_uri'};
```

Comment:
I think we have config on $self:
$self->config
Could you check it, please?
So, possibly we don't need these lines (line 8-12)